sdks/python/apache_beam/utils/subprocess_server.py

```diff
@@ -0,0 +1,209 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+
+import logging
+import os
+import shutil
+import signal
+import socket
+import subprocess
+import tempfile
+import threading
+import time
+
+import grpc
+from future.moves.urllib.error import URLError
+from future.moves.urllib.request import urlopen
+
+from apache_beam.version import __version__ as beam_version
+
+
+class SubprocessServer(object):
+  """An abstract base class for running GRPC Servers as an external process."""
+  def __init__(self, stub_class, cmd, port=None):
```

Comment:
Based on your last response I looked at this code more closely.  I understand it now (I think), but it wouldn't hurt to document these `__init__` args, if you're up for it. 

As I understand it, `cmd` must start a gRPC servicer, and `stub_class` must be compatible with that servicer.

Reply:
Precisely. Thanks for the encouragement for more documentation :). Added.